src/Nethermind/Nethermind.Blockchain/Processing/BlockProcessor.BlockProductionTransactionPicker.cs

```diff
@@ -95,7 +95,7 @@ private bool HasEnoughFounds(Transaction transaction, UInt256 senderBalance, Add
                     return false;
                 }
 
-                if (transaction.IsEip1559 && !transaction.IsServiceTransaction && senderBalance < (UInt256)transaction.GasLimit * transaction.MaxFeePerGas)
+                if (eip1559Enabled && !transaction.IsServiceTransaction && senderBalance < (UInt256)transaction.GasLimit * transaction.MaxFeePerGas + transaction.Value)
```

Comment:
This and the check above in line 90 look similar (same?) as the ones in TransactionProcessor, should we have an extension method or something?